addons/website_event/controllers/main.py

```diff
@@ -415,14 +417,44 @@ def get_formated_date(self, event):
         month = babel.dates.get_month_names('abbreviated', locale=get_lang(event.env).code)[start_date.month]
         return ('%s %s%s') % (month, start_date.strftime("%e"), (end_date != start_date and ("-" + end_date.strftime("%e")) or ""))
 
-    def _extract_searched_event_tags(self, searches):
+    def _extract_searched_event_tags(self, searches, slug_tags):
         tags = request.env['event.tag']
-        if searches.get('tags'):
-            try:
-                tag_ids = literal_eval(searches['tags'])
-            except:
-                pass
-            else:
-                # perform a search to filter on existing / valid tags implicitely + apply rules on color
-                tags = request.env['event.tag'].search([('id', 'in', tag_ids)])
+        if slug_tags:
+            tags = self._event_search_tags_slug(slug_tags)
+        elif 'tags' in searches:
+            tags = self._event_search_tags_ids(searches['tags'])
         return tags
+
+    def _slugify_tags(self, tag_ids, toggle_tag_id=None):
+        """ Prepares a comma separated slugified tags for the sake of readable URLs.
+        :param toggle_tag_id: add the tag being clicked (current_tag) to the already
+          selected tags (tag_ids) as well as in URL; if tag is already selected
+          by the user it is removed from the selected tags (and so from the URL);
+        """
+        tag_ids = list(tag_ids)
+        if toggle_tag_id and toggle_tag_id in tag_ids:
+            tag_ids.remove(toggle_tag_id)
+        elif toggle_tag_id:
+            tag_ids.append(toggle_tag_id)
+
+        return ','.join(request.env['ir.http']._slug(tag_id) for tag_id in request.env['event.tag'].browse(tag_ids)) or ''
+
+    def _event_search_tags_ids(self, search_tags):
+        """ Input: %5B4%5D """
```

Comment:
Well, during testing, I found out that this will select the particular tag from the relevant dropdown but not return results based on it; instead, it returns all event records to be rendered. 

Can you please have another look at the retrospective compatibility case?

After that, we are almost at the conclusion and good to go.

Also, Change the commit message formate in Purpose - Specs manner.

Thanks

Reply:
@spe-odoo Thanks for pointing it out resolved the issue would you mind taking a look at it again and let me know if i missed something Thanks ;)